### ISSUE TITLE
multilingual support for navigation bar

### ISSUE THREAD

**Issue by cravussin** [getgrav/grav-theme-agency#11]
Hello,

It seems that the theme doesn't support multilingual content, i.e. the navigation bar items (about, portfolio, etc...) cannot be displayed in another language. Any tips to work around that?

**Comment by flaviocopes** [comment]
Did you read how multilanguage works in Grav? https://learn.getgrav.org/content/multi-language 

You need to setup language-specific pages, then their title will be used in the menu.

**Comment by cravussin** [comment]
Yes I read the documentation. In the agency theme the links that show up in the navbar are defined in  site.yaml, under the "links:" section. 

The site.yaml file does not support multilingual, that's why I opened  an issue.

**Comment by flaviocopes** [comment]
Oh I see they are on-page links.

It's not built with multilang in mind, you can do that by having in site.yaml

```
```

and edit themes/agency/templates/partials/navigation.html.twig like

```
```

**Comment by flaviocopes** [comment]
Oh I see they are on-page links, not real pages.

It's not built with multilang in mind, you can do that by having in site.yaml

```

links:
  en:
    - title: MY_TRANSLATION_STRING
      url: '#services'
```

and edit themes/agency/templates/partials/navigation.html.twig line 29 so it translates the string, like

```
{{ link.title|t }}
```

instead of  `{{ link.title }}`.

Then add that string translations in your `user/languages/` (https://learn.getgrav.org/content/multi-language#translation-overrides)

**Comment by cravussin** [comment]
I'll try that and let you know, many thanks for the info.

**Comment by cravussin** [comment]
Unfortunately it doesn't seem to be working, all the items in the navbar have disappeared ..
Here's the modified line is navigation.html.twig:

`<li><a class="page-scroll" href="{{ domain }}{{ link.url }}" {% if link.url | contains('http') %}target="_blank"{% endif %}>{{ link.title|t }}</a></li>`

Here's the links section of the site.yaml file:

links:
  fr:
    - title: Avantages
      url: '#avantages'
    - title: Société
      url: '#societe'
    - title: Team
      url: '#team'
    - title: Contact
      url: '#contact'
  en:
    - title: Advantages
      url: '#avantages
    - title: Company
      url: '#societe'
    - title: Team
      url: '#team'
    - title: Contact
      url: '#contact'

It doesn't work with translation overrides neither.

**Comment by flaviocopes** [comment]
That's not what I suggested. That snippet you posted is not supposed to work. `link.title` is not existing. Use `{{ dump(link) }}` to check its content in the debug bar https://learn.getgrav.org/advanced/debugging#debug-bar

Also, the structure of the YAML is not right. 

```
links:
  fr:
    title: Avantages
    url: '#avantages'
```

should do.

You would have `link.fr.title` maybe, which needs to be accessed in a different way in Twig, something like `{{ link[grav.language.language].title }}`

**Comment by cravussin** [comment]
I managed to make it work by using per language twig templates (templates/en/partials/navigation.html.twig) and using

{% for link in site.links.en %}

and 

{% for link in site.links.fr %} in the default template

I removed the translation filters.

Not the most elegant stuff but hey it works! 

-- 
On 30 juin 2016 at 13:12:14, Flavio Copes (notifications@github.com) wrote:

That's not what I suggested. That snippet you posted is not supposed to work. link.title is not existing. Use {{ dump(link) }} to check its content in the debug bar https://learn.getgrav.org/advanced/debugging#debug-bar

Also, the structure of the YAML is not right.

links:
  fr:
    title: Avantages
    url: '#avantages'

should do.

You would have link.fr.title maybe, which needs to be accessed in a different way in Twig, something like {{ link[grav.language.language].title }}
